stringtemplate3/src/test/java/org/jdbi/v3/sqlobject/BindBeanListTest.java

```diff
@@ -0,0 +1,217 @@
+    {
+        final Jdbi dbi = db.getJdbi();
+        dbi.installPlugin(new SqlObjectPlugin());
+        dbi.registerRowMapper(new SomethingMapper());
+        handle = dbi.open();
+
+        handle.execute("insert into something(id, name) values(1, '1')");
+        handle.execute("insert into something(id, name) values(2, '2')");
+
+        // "control group" element that should *not* be returned by the queries
+        handle.execute("insert into something(id, name) values(3, '3')");
+    }
+
+    @AfterClass
+    public static void exit()
+    {
+        handle.close();
+    }
+
+    //
+
+    @Test
+    public void testSomethingWithExplicitAttributeName()
+    {
+        final SomethingWithExplicitAttributeName s = handle.attach(SomethingWithExplicitAttributeName.class);
+
+        final List<Something> out = s.get(
+                new SomethingKey(1, "1"),
+                new SomethingKey(2, "2"));
+
+        assertThat(out).hasSize(2);
+    }
+
+    @UseStringTemplateStatementRewriter
+    public interface SomethingWithExplicitAttributeName {
+        @SqlQuery("select id, name from something where (id, name) in (<keys>)")
+        List<Something> get(@BindBeanList(value = "keys", propertyNames = {"id", "name"})
+                                    SomethingKey... blarg);
+    }
+
+    //
+
+    @Test
+    public void testSomethingByVarargsWithVarargs()
+    {
+        final SomethingByVarargs s = handle.attach(SomethingByVarargs.class);
+
+        final List<Something> out = s.get(
+                new SomethingKey(1, "1"),
+                new SomethingKey(2, "2"));
+
+        assertThat(out).hasSize(2);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSomethingByVarargsWithEmptyVarargs()
+    {
+        final SomethingByVarargs s = handle.attach(SomethingByVarargs.class);
+
+        final List<Something> out = s.get();
+    }
+
+    @UseStringTemplateStatementRewriter
+    public interface SomethingByVarargs
+    {
+        @SqlQuery("select id, name from something where (id, name) in (<keys>)")
+        List<Something> get(@BindBeanList(propertyNames = {"id", "name"}) SomethingKey... keys);
+    }
+
+    //
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSomethingByArrayWithNull()
+    {
+        final SomethingByArray s = handle.attach(SomethingByArray.class);
+
+        s.get(null);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testSomethingByArrayWithEmptyArray()
+    {
+        final SomethingByArray s = handle.attach(SomethingByArray.class);
+
+        s.get(new SomethingKey[]{});
+    }
+
+    @Test
+    public void testSomethingByArrayWithNonEmptyArray()
+    {
+        final SomethingByVarargs s = handle.attach(SomethingByVarargs.class);
+
+        final List<Something> out = s.get(
+                new SomethingKey(1, "1"),
+                new SomethingKey(2, "2"));
+
+        assertThat(out).hasSize(2);
```

Comment:
I was cribbing off `BindListTest`, which only had the size assertions. I'll add result assertions to both places.